MANUAL.md

```diff
@@ -0,0 +1,99 @@
+# Prometheus Genesis Kit Manual 
+The Prometheus Genesis Kit deploys a single-VM installation of
+Prometheus, Grafana, and AlertManager. It's capable of monitoring BOSH
+and Cloudfoundry.
+
+## Requirements
+
+[Node Exporter][1] must be installed on all BOSH VMs you would like
+to monitor. Please consult the README of that repository on how to
+setup the BOSH addon.
+
+
+## Features
+
+## SSL Certificates
+
+* `self-signed-certs` - If you wish to have Genesis generate
+  self-signed certs for you. 
+* `provided-cert` - If you have SSL cert/key to provide, which is
+  grabbed from Vault via path:
+  `$GENESIS_VAULT_PATH/nginx/ssl_certificate:certificate` and
+  `$GENESIS_VAULT_PATH/nginx/ssl_certificate:key`
+
+## Authentication
+
+* `http-auth` - By default, only Grafana supports authentication via
+  the web UI. This feature adds HTTP Basic Authentication to
+  AlertManager and Prometheus with automatically generated
+  credentials.
+
+## Monitoring
+
+* `monitor-cf` - Have Prometheus connect to the CF Firehose to track
+  CF app status + more. Requires two UAA accounts, one with the
+  `cloud_controller.admin_read_only` scope, and the `doppler.firehose`
+  scope. To create these accounts, here's an example:
+```
+uaac client add prometheus-cf \
+  --name prometheus-cf \
+  --secret <64 char secret> \
+  --authorized_grant_types client_credentials,refresh_token \
+  --authorities cloud_controller.admin_read_only
+
+uaac client add prometheus-firehose \
+  --name prometheus-firehose \
+  --secret <64 char secret> \
+  --authorized_grant_types client_credentials,refresh_token \
+  --authorities doppler.firehose
+```
+
+  This will add the user `prometheus-cf` and `prometheus-firehose`,
+  with the secrets of your choice. `prometheus-cf` will have the
+  `cloud_controller.admin_read_only` scope, and the
+  `prometheus-firehose` will have the `doppler.firehose` scope.
+
+  These credentials are stored in `$GENESIS_VAULT_PATH/cf_uaa_logins`:
```

Comment:
Do we need two?  Can we get by with one, that has all of the scopes?

Reply:
I was originally on the side of single account, but a few people mentioned principle of least privilege. I planned to alter the CF kit to automatically create these credentials anyway, so most operators wouldn't have to deal with this anyway.

Reply:
I'm not sure I agree with the principle of least privilege here.  Worst case, you build a `prometheus` client strictly for auditing purposes.  But it's the same system doing the queries against cf and the firehose so ...